yt_dlp/extractor/plex.py

```diff
@@ -0,0 +1,390 @@
+    _CLIENT_IDENTIFIER = '26275d8b-63b3-49f9-05a3-6c8659c00413'  # can get in cookie
+
+    def _perform_login(self, username, password):
+        self.write_debug('Trying to login')
+        try:
+            resp_api = self._download_json(
+                'https://plex.tv/api/v2/users/signin', 'Auth', query={'X-Plex-Client-Identifier': self._CLIENT_IDENTIFIER},
+                data=f'login={username}&password={password}&rememberMe=true'.encode(),
+                headers={'Accept': 'application/json'}, expected_status=429,
+                note='Downloading JSON Auth Info')
+            self._TOKEN = resp_api.get('authToken') or self._TOKEN
+            self.write_debug('login successfully')
+        except ExtractorError as e:
+            self.write_debug(f'There\'s error on login : {e.cause}, trying to use non-login method')
+
+    def _get_formats_and_subtitles(self, selected_media, display_id, sites_type='vod', metadata_field={}):
+        formats, subtitles = [], {}
+        fmt, subs = [], {}
+        if isinstance(selected_media, str):
+            selected_media = [selected_media]
+        for media in selected_media:
+            if determine_ext(media) == 'm3u8' or media.endswith('hls'):
+                fmt, subs = self._extract_m3u8_formats_and_subtitles(
+                    f'{self._CDN_ENDPOINT[sites_type]}{media}',
+                    display_id, query={'X-PLEX-TOKEN': self._TOKEN})
+
+            elif determine_ext(media) == 'mpd':
+                fmt, subs = self._extract_mpd_formats_and_subtitles(
+                    f'{self._CDN_ENDPOINT[sites_type]}{media}',
+                    display_id, query={'X-PLEX-TOKEN': self._TOKEN},)
+
+            else:
+                formats.append({
+                    'url': f'{self._CDN_ENDPOINT[sites_type]}{media}?X-Plex-Token={self._TOKEN}',
+                    'ext': 'mp4',
+                    **metadata_field
+                })
+
+            formats.extend(fmt)
+            self._merge_subtitles(subs, target=subtitles)
+
+        return formats, subtitles
+
+    def _get_clips(self, nextjs_json, display_id):
+        entries = []
+        self.write_debug('Trying to download Extras/trailer')
+
+        media_json = self._download_json(
+            'https://play.provider.plex.tv/playQueues', display_id,
+            query={'uri': f'provider://tv.plex.provider.vod{nextjs_json["Extras"]["key"]}'}, data=b'',
+            headers={'X-PLEX-TOKEN': self._TOKEN, 'Accept': 'application/json'})
+
+        for media in media_json['MediaContainer']['Metadata']:
+            for media_ in traverse_obj(media, ('Media', ..., 'Part', ..., 'key')):
+                fmt, sub = self._get_formats_and_subtitles(media_, display_id)
+                new_fmt = []
+                for fmt_ in fmt:
+                    fmt_.update({'id': media['ratingKey'], 'title': media.get('title')})
+                    new_fmt.append(fmt_)
+                entries.extend(new_fmt)
+
+        return entries
```

Comment:
This function extract all video as single video. I mean the same video with different resolution treated as different video in `_extract_m3u8_formats_and_subtitles`. I don't know why this happen. 

The api is kind of weird too. The api return something with suffix `mp4` (eg: `https://video.internetvideoarchive.net/video.mp4?customerid=112548&cmd=6&e=253370764800&fmt=11&publishedid=579373&h=d239a70898d675940068c66aaf3340aa`) but return `m3u8`. The `m3u8` is a bit different too. This issue only happen in Extras.

Here's the sample of m3u8:
```
#EXTM3U
#EXT-X-VERSION:3
## Created with Unified Streaming Platform(version=1.8.4)

# variants
#EXT-X-STREAM-INF:BANDWIDTH=1787000,CODECS="mp4a.40.2,avc1.4D401F",RESOLUTION=720x405
https://d2em1d4z1elu3k.cloudfront.net/iva-svideo/3eb9-583256/583256.secure.ism/583256.secure-audio=256913-video=1428000.m3u8?c=112548&sub=not-set&ser=637902128180000000
#EXT-X-STREAM-INF:BANDWIDTH=333000,CODECS="mp4a.40.2,avc1.4D401F",RESOLUTION=320x180
https://d2em1d4z1elu3k.cloudfront.net/iva-svideo/3eb9-583256/583256.secure.ism/583256.secure-audio=128706-video=185000.m3u8?c=112548&sub=not-set&ser=637902128180000000
#EXT-X-STREAM-INF:BANDWIDTH=711000,CODECS="mp4a.40.2,avc1.4D401F",RESOLUTION=320x180
https://d2em1d4z1elu3k.cloudfront.net/iva-svideo/3eb9-583256/583256.secure.ism/583256.secure-audio=256913-video=413000.m3u8?c=112548&sub=not-set&ser=637902128180000000
#EXT-X-STREAM-INF:BANDWIDTH=998000,CODECS="mp4a.40.2,avc1.4D401F",RESOLUTION=480x270
https://d2em1d4z1elu3k.cloudfront.net/iva-svideo/3eb9-583256/583256.secure.ism/583256.secure-audio=256913-video=684000.m3u8?c=112548&sub=not-set&ser=637902128180000000
#EXT-X-STREAM-INF:BANDWIDTH=156000,CODECS="mp4a.40.2,avc1.4D401F",RESOLUTION=320x180
https://d2em1d4z1elu3k.cloudfront.net/iva-svideo/3eb9-583256/583256.secure.ism/583256.secure-audio=80446-video=66000.m3u8?c=112548&sub=not-set&ser=637902128180000000
#EXT-X-STREAM-INF:BANDWIDTH=2909000,CODECS="mp4a.40.2,avc1.4D401F",RESOLUTION=1280x720
https://d2em1d4z1elu3k.cloudfront.net/iva-svideo/3eb9-583256/583256.secure.ism/583256.secure-audio=256913-video=2487000.m3u8?c=112548&sub=not-set&ser=637902128180000000
#EXT-X-STREAM-INF:BANDWIDTH=3775000,CODECS="mp4a.40.2,avc1.4D401F",RESOLUTION=1280x720
https://d2em1d4z1elu3k.cloudfront.net/iva-svideo/3eb9-583256/583256.secure.ism/583256.secure-audio=256913-video=3304000.m3u8?c=112548&sub=not-set&ser=637902128180000000
#EXT-X-STREAM-INF:BANDWIDTH=4680000,CODECS="mp4a.40.2,avc1.4D401F",RESOLUTION=1920x1080
https://d2em1d4z1elu3k.cloudfront.net/iva-svideo/3eb9-583256/583256.secure.ism/583256.secure-audio=256913-video=4158000.m3u8?c=112548&sub=not-set&ser=637902128180000000
#EXT-X-STREAM-INF:BANDWIDTH=5558000,CODECS="mp4a.40.2,avc1.4D401F",RESOLUTION=1920x1080
https://d2em1d4z1elu3k.cloudfront.net/iva-svideo/3eb9-583256/583256.secure.ism/583256.secure-audio=256913-video=4986000.m3u8?c=112548&sub=not-set&ser=637902128180000000
```

Reply:
Fixed in https://github.com/yt-dlp/yt-dlp/pull/4366/commits/ae7a2616352dc99198b5b55aa6ec51fdf0076b74